### PR TITLE
 Added Availability workflows

### PR DESCRIPTION
**Pull request by nevil-mathew**

### CODE REVIEW

File: src/controllers/v1/availability.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	 * Create availability.
+	 * @method
+	 * @name create
+	 * @param {Object} req - The request object containing the request body and decoded token.
+	 * @returns {Promise<Object>} - The created availability object.
+	 */
+	async create(req) {
+		try {
+			req.decodedToken = {
+				id: 133,
```
Comment (rakeshSgr) [inline src/controllers/v1/availability.js:18]: @nevil-mathew why hardcoded the id?

Comment (nevil-mathew) [reply src/controllers/v1/availability.js:18]: For testing purposes, will remove these and update.

File: src/database/models/availabilities.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				allowNull: false,
+				autoIncrement: true,
+				primaryKey: true,
+				type: DataTypes.INTEGER,
+			},
+			user_id: {
+				allowNull: false,
+				type: DataTypes.INTEGER,
+			},
+			event_name: { type: DataTypes.STRING, allowNull: false },
```
Comment (rakeshSgr) [inline src/database/models/availabilities.js:16]: @nevil-mathew how the event will be created?

File: src/dtos/availability.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+					}
+					occurrenceDate.add(
+						userAvailability.repeat_increment || 1,
+						userAvailability.repeat_unit.toLowerCase()
+					)
+					occurrenceEndDate.add(
+						userAvailability.repeat_increment || 1,
+						userAvailability.repeat_unit.toLowerCase()
+					)
+					if (
```
Comment (rakeshSgr) [inline src/dtos/availability.js:96]: @nevil-mathew if repeat_unit is year then?

File: src/controllers/v1/availability.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ */
+module.exports = class Availability {
+	/**
+	 * Create availability.
+	 * @method
+	 * @name create
+	 * @param {Object} req - The request object containing the request body and decoded token.
+	 * @returns {Promise<Object>} - The created availability object.
+	 */
+	async create(req) {
```
Comment (rakeshSgr) [inline src/controllers/v1/availability.js:15]: @nevil-mathew update the postman collection also

Comment (rakeshSgr) [inline src/controllers/v1/availability.js:15]: @nevil-mathew update the postman collection also

File: src/database/models/availabilities.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+'use strict'
+module.exports = (sequelize, DataTypes) => {
+	const Availability = sequelize.define(
+		'Availability',
+		{
+			id: {
```
[the remark this answers is not in the document]
Comment (nevil-mathew) [reply src/database/models/availabilities.js:6]: As discussed booking wont be part of this work flows

File: src/dtos/availability.js (hunk 3 above)
Comment (nevil-mathew) [reply src/dtos/availability.js:96]: Then we'll add an year to the current date

File: src/database/models/availabilities.js (hunk 2 above)
Comment (nevil-mathew) [reply src/database/models/availabilities.js:16]: Via CURD apis

File: src/controllers/v1/availability.js (hunk 4 above)
Comment (nevil-mathew) [reply src/controllers/v1/availability.js:15]: updated

Comment (nevil-mathew) [reply src/controllers/v1/availability.js:15]: updated

File: src/dtos/availability.js (hunk 3 above)
Comment (rakeshSgr) [reply src/dtos/availability.js:96]: is that logic implemented?

**Pull request merged**